Program.cs

```diff
@@ -0,0 +1,249 @@
+                Console.Error.WriteLine("Error: No repository urls provided.");
+                app.ShowHelp();
+                return 1;
+            }
+
+            if (doCommit && string.IsNullOrWhiteSpace(commitMessage))
+            {
+                Console.Error.WriteLine("Error: Specify a commit message with --commit-message when using --commit or --pull-request");
+                app.ShowHelp();
+                return 1;
+            }
+
+            if (createPr && string.IsNullOrWhiteSpace(changesBranchName))
+            {
+                Console.Error.WriteLine("Error: Specify a branch to create with --changes-branch when using --pull-request");
+                app.ShowHelp();
+                return 1;
+            }
+
+            var credentials = GetCredentials();
+
+            foreach (var repoUrl in urls)
+            {
+                RepositoryProcessor processor = null;
+                try
+                {
+                    processor = new RepositoryProcessor(repoUrl, credentials);
+
+                    if (!string.IsNullOrWhiteSpace(baseBranchName))
+                    {
+                        processor.CheckoutBaseBranch(baseBranchName);
+                    }
+
+                    if (!string.IsNullOrWhiteSpace(changesBranchName))
+                    {
+                        processor.CreateChangesBranch(changesBranchName);
+                    }
+
+                    processor.RunCommand(commandToRun);
+
+                    if (!doCommit || !processor.HasChanges)
+                    {
+                        continue;
+                    }
+
+                    processor.Commit(commitMessage);
+
+                    if (push)
+                    {
+                        processor.Push();
+                    }
+
+                    if (createPr)
+                    {
+                        processor.CreatePullRequest(commitMessage);
+                    }
+                }
+                finally
+                {
+                    // Wether success or failure, clean up the cloned repository
+                    if (processor != null)
+                    {
+                        processor.Cleanup();
+                    }
+                }
+            }
+
+            return 0;
+        }
+
+        private static UsernamePasswordCredentials GetCredentials()
+        {
+            Console.Write("Enter git username: ");
+            var userName = Console.ReadLine();
+            Console.Write("Enter git password: ");
+            var password = GetHiddenConsoleInput();
+            Console.WriteLine();
+
+            var credentials = new UsernamePasswordCredentials() { Username = userName, Password = password };
+            return credentials;
+        }
+
+        private static string GetHiddenConsoleInput()
+        {
+            StringBuilder input = new StringBuilder();
+            while (true)
+            {
+                var key = Console.ReadKey(true);
```

Comment:
If we're adding/removing by character wouldn't this be a perfect candidate for using a SecureString https://docs.microsoft.com/en-us/dotnet/api/system.security.securestring?view=netcore-2.2

Reply:
May not be possible if other libraries don't support this.

Reply:
Libgit2sharp does not support secure strings unfortunately.